### ISSUE TITLE
beforeRemove event listener also affecting regular navigate() action

### ISSUE THREAD

**Issue by KrisLau** [react-navigation/react-navigation#10088]
### Current behavior

When I follow [this documentation](https://reactnavigation.org/docs/preventing-going-back/) for preventing the user from going back, it also seems to be preventing me from doing a navigation.navigate() action.

### Expected behavior

Upon swiping left on iOS, it should redirect me to settings screen but instead I cant seem to navigate anywhere.

### Reproduction

https://snack.expo.dev/5EcYJd2ZJ

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [ ] @react-navigation/native-stack

### Environment

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |    ^5.9.4
| @react-navigation/stack                |     ^5.14.4
| react-native                           |      ^0.64.2

**Comment by KrisLau** [comment]
Sorry, looks like its working now!

**Comment by KrisLau** [comment]
So weird, I haven't changed any of the code but now it's affecting the navigate action again. I tried reloading the app and after the app reloaded, the `useEffect` worked correctly without affecting the navigate action. I did a couple more reloads and it's still working correctly.  

I'm just going to leave this issue in case anyone else if experiencing this as well and might have better insight into what's causing it

**Comment by AlixH** [comment]
Same issue here

**Comment by fabriciosautner** [comment]
same issue here

**Comment by rifqiabrory** [comment]
same issue here

**Comment by satya164** [comment]
> it also seems to be preventing me from doing a navigation.navigate() action

Your `navigation.navigate` is doing the same thing as going back - it's going to a previous screen. As documented (and per the name `beforeRemove`), any action that removes the screen will trigger this listener.

**Comment by Real-Musafir** [comment]
So what will be the way to navigate to another screen in the stack instead of going back ?

**Comment by fengbujue2022** [comment]
any update?

**Comment by Satellite-system** [comment]
I'm facing the same issue does anyone got to solve it ?

**Comment by KrisLau** [comment]
@fengbujue2022 @Satellite-system 

> > it also seems to be preventing me from doing a navigation.navigate() action
> 
> Your `navigation.navigate` is doing the same thing as going back - it's going to a previous screen. As documented (and per the name `beforeRemove`), any action that removes the screen will trigger this listener.

@fengbujue2022 @Satellite-system There is no solution per se since this is the expected behaviour

**Comment by PragyaSingla** [comment]
I am navigating to next screen but it is going to other (as specified in beforeRemove). Why instead of following navigate event, it is listening to beforeRemove.
Anyone has any solution to this problem?

**Comment by autoclickvn** [comment]
Sorry my English might not be very good.

I was facing the same issue and realized that when using navigate (or goBack) it somehow goes back (underground operations) to the front screen and the `beforeRemove` event is fired again. I followed up on the docs [https://reactnavigation.org/docs/preventing-going-back/#limitations](https://reactnavigation.org/docs/preventing-going-back/#limitations)

_**This event is not triggered when a screen is being unfocused but not removed. For example:

The user pushed a new screen on top of the screen with the listener in a stack.**_

I have tried pushing the screens to the Stack (using StackActions) and this works for me. Hope this helps everyone.
